registrations/views.py

```diff
@@ -166,14 +166,22 @@ def post(self, request):
             api_url=settings.IDENTITY_STORE_URL,
             auth_token=settings.IDENTITY_STORE_TOKEN
         )
-        # load the users sources - posting users should only have one source
-        source = Source.objects.get(user=self.request.user)
         serializer = ThirdPartyRegistrationSerializer(data=request.data)
         if serializer.is_valid():
             mom_msisdn = serializer.validated_data['mom_msisdn']
             hcw_msisdn = serializer.validated_data['hcw_msisdn']
             lang_code = serializer.validated_data['mom_lang']
             lang_code = transform_language_code(lang_code)
+            authority = serializer.validated_data['authority']
+            # load the users sources with authority mapping
+            if authority == 'chw':
+                source_auth = 'hw_limited'
```

Comment:
I got limited from here: https://github.com/praekeltfoundation/ndoh-hub/blob/feature%2Fextend-source-lookup/registrations/models.py#L19 which is right?